vip-integration-helper/integrations/integrations.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Automattic\VIP\Integrations;
+
+defined( 'ABSPATH' ) || die();
+
+require_once __DIR__ . '/integration.php';
+
+/**
+ * Class used to track and activate registered integrations.
+ */
+class Integrations {
+	/**
+	 * Collection of registered integrations.
+	 *
+	 * @var array<Integration>
+	 */
+	private $integrations = array();
+
+	/**
+	 * Return singleton instance of class.
+	 */
+	public static function instance(): Integrations {
+		static $instance = null;
+
+		if ( null === $instance ) {
+			$instance = new self();
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Registers an integration.
+	 *
+	 * @param string       $slug        A unique identifier for the integration.
+	 * @param class-string $class_name Fully-qualified class name that will be instantiated.
```

Comment:
I stole this from [wp-parsely's `Integrations` implementation](https://github.com/Parsely/wp-parsely/blob/2f5c3428c7c11c6177c5622a2477022072e5c2b3/src/Integrations/class-integrations.php#L53):

```php
/**
 * ...
 * @param class-string|Integrations $class_or_object Fully-qualified class name, or an instantiated object.
 *                                                   If a class name is passed, it will be instantiated.
```

but admittedly it's the first time I've seen it used. It appears to be [a PHPStan/PHPDoc type](https://phpstan.org/writing-php-code/phpdoc-types#class-string) and we also use it in a few places in mu-plugins code as well as wp-parsely code. Should I still replace this with `string`?

Reply:
If PHPStan isn't being used against this code, then yes. Otherwise leave it as is (I'm curious how IDEs interpret it).

Reply:
Changed this to `string` since we're not using PHPStan directly in mu-plugins.